src/utils/firebase.js

```diff
@@ -45,3 +45,20 @@ export const login = async () => {
     return false;
   }
 };
+
+export const getUserFromUid = async (uid, callback) => {
```

Comment:
i think its best to have the logic for setting the user inside the scan page, just to keep everything together. It’s probably best if this either returns a user ref or null

Reply:
actually since we already pulled the data just return the data if a user exists

Reply:
For some reason it would return null if I didn't use the callback. Oh well, its working in Scan now lol